pytorch_lightning/trainer/training_loop.py

```diff
@@ -608,7 +608,12 @@ def run_training_batch(self, batch, batch_idx, dataloader_idx):
         batch_outputs = [[] for _ in range(len(optimizers))]
 
         if batch is None:
-            return AttributeDict(signal=0, grad_norm_dic=grad_norm_dic)
+            self.warning_cache.warn("train_dataloader yielded None. If this was on purpose, ignore this warning...")
+            return AttributeDict(
+                signal=0,
+                grad_norm_dic=grad_norm_dic,
+                training_step_output_for_epoch_end=batch_outputs,
+            )
```

Comment:
what happens in distributed training? won't we go out of sync across ranks if the batch is None on some ranks and non-None on others?

Reply:
I'm not sure. It will be closer to working that it was before, but this use case may still need some work for distributed. The new behaviour (that is, after this fix) should be the same as when returning None from a step

Reply:
There are two options:
- either the trainer runs an all reduce at each step to determine whether all ranks skip. This is a general solution but wastes perf for any users who won't skip batches
- the user runs the all reduce in their lightning module and determines when to skip. This way it's specific to just modules which need to skip batches or steps for any reason.

I prefer the latter approach because I don't want to slow down the trainer for everyone to handle a few corner cases

Reply:
Yeah, I think it's fine to make the user responsible for making all ranks skip in sync if they want distributed skipping

Reply:
Returning `None` in any kind of distributed setting is currently unsupported. We have this (very old) PR to add support but it's currently frozen

#5359

You can have a look at the proposal though